crates/solvers/src/boundary/baseline.rs

```diff
@@ -44,47 +44,43 @@ impl<'a> Solver<'a> {
             max_hops,
         );
 
-        let segments = match request.side {
-            order::Side::Buy => {
-                let (segments, _) = candidates
-                    .iter()
-                    .filter_map(|path| {
-                        let estimate = baseline_solver::estimate_sell_amount(
-                            request.buy.amount,
-                            path,
-                            &self.amms,
-                        )?;
-                        let segments = self.traverse_path(
-                            &estimate.path,
-                            request.sell.token.0,
-                            estimate.value,
-                        )?;
-
-                        let sell = estimate.value;
-                        let buy = segments
-                            .last()
-                            .map(|segment| segment.output.amount)
-                            .unwrap_or(sell);
-
-                        if sell <= request.sell.amount && buy >= request.buy.amount {
-                            Some((segments, sell))
-                        } else {
-                            None
-                        }
-                    })
-                    .min_by_key(|(_, sell)| *sell)?;
-                segments
-            }
-            order::Side::Sell => {
-                let estimate = candidates
-                    .iter()
-                    .filter_map(|path| {
-                        baseline_solver::estimate_buy_amount(request.sell.amount, path, &self.amms)
-                    })
-                    .filter(|estimate| estimate.value >= request.buy.amount)
-                    .max_by_key(|estimate| estimate.value)?;
-                self.traverse_path(&estimate.path, request.sell.token.0, request.sell.amount)?
-            }
+        let (segments, _) = match request.side {
+            order::Side::Buy => candidates
+                .iter()
+                .filter_map(|path| {
+                    let estimate = baseline_solver::estimate_sell_amount(
+                        request.buy.amount,
+                        path,
+                        &self.amms,
+                    )?;
+                    let segments =
+                        self.traverse_path(&estimate.path, request.sell.token.0, estimate.value)?;
+                    let buy = segments
+                        .last()
+                        .map(|segment| segment.output.amount)
+                        .unwrap_or(estimate.value);
+
+                    (estimate.value <= request.sell.amount && buy >= request.buy.amount)
+                        .then_some((segments, estimate))
+                })
+                .min_by_key(|(_, estimate)| estimate.value)?,
```

Comment:
nit: I liked that you called the estimate either `sell` or `buy` in the previous version.
That made it trivial to understand why `min_by_key()` was correct in one case and `max_by_key()` was correct in the other case.

Reply:
218adf851d76f18c6a420c0f2cbd1ba30a00c70e